### PR TITLE
Avoid saving large non-string data

### PR DESCRIPTION
**Pull request by hluk**
JIRA: RHELWF-558
Signed-off-by: Lukas Holecek <hluk@email.cz>

### CODE REVIEW

**Comment by hluk** [comment]
We should avoid passing dict values in result data. ResultsDB [simply converts dict to string](https://pagure.io/taskotron/resultsdb/blob/9250ae962b96ac00e0160e80af15b4adf4750d66/f/resultsdb/controllers/api_v2.py#_692).

File: resultsdbupdater/utils.py [hunk 1]
```diff
@@ -18,11 +18,17 @@ def crop_data(log, data):
     Even thought size for result data are not limited in database schema,
     Postgresql index has a size limited ("Values larger than 1/3 of a buffer
     page cannot be indexed").
+
+    Raises InvalidMessageError if non-string value is too large.
     """
     for k, v in data.items():
         if isinstance(v, str) and len(v) > MAX_RESULT_DATA_SIZE:
```
Comment (mvalik) [inline resultsdbupdater/utils.py:25]:
```
if len(str(v)) > MAX_RESULT_DATA_SIZE:
  if isinstance(v, str):
   ...
  elif not isinstance(v, list) or any(len(str(item)) > MAX_RESULT_DATA_SIZE for item in v):
  ...
```

Comment (hluk) [reply resultsdbupdater/utils.py]: I changed it, so it's more clear.

**Comment by gnaponie** [comment]
Looks good to me +1

**Pull request merged**